INSERT column sorting is broken

https://github.com/go-xorm/builder/commit/395bcf3d8aa42863b8062dc552a002ca0dc71c35 adds sorting to the builder.Insert function, but sorts incorrectly, resulting in mangled column/value mapping (e.g. an `Insert(builder.Eq{"y":0, "x":1})` may result in `INSERT INTO foo (x, y) VALUES (0, 1)` instead of the correct `(1, 0)`).

https://github.com/go-xorm/builder/blob/ef80bec6771b66907d34a878fe8b3914b9f5cda8/builder.go#L278-L281

Due to how the `"sort"` package works, `Swap()` and `Less()` are called interleaved - using `sort.Slice()` where the `Less()` function uses the *original* order of the slice won't work - it needs to use the *current* order.

See https://play.golang.org/p/ITcqKVBvdbW for an example of how this goes wrong - both `keys` and `values` are in the same order to start with, but due to Less() using the original order, the sorting puts the `values` slice in a semi-random order dependent on the original order of the `keys` slice. Uncommenting line 26, i.e. keeping the keys and values slices in the same order as the sorting algorithm progresses, fixes the sort (and makes the `sort.Ints()` call redundant as a bonus!)

Example failing test:
```diff
diff --git a/builder_insert_test.go b/builder_insert_test.go
index 5ea8662..ef3db52 100644
--- a/builder_insert_test.go
+++ b/builder_insert_test.go
@@ -15,6 +15,10 @@ func TestBuilderInsert(t *testing.T) {
        assert.NoError(t, err)
        assert.EqualValues(t, "INSERT INTO table1 (c,d) Values (1,2)", sql)
 
+       sql, err = Insert(Eq{"e": 3}, Eq{"c": 1}, Eq{"d": 2}).Into("table1").ToBoundSQL()
+       assert.NoError(t, err)
+       assert.EqualValues(t, "INSERT INTO table1 (c,d,e) Values (1,2,3)", sql)
+
        sql, err = Insert(Eq{"c": 1, "d": Expr("SELECT b FROM t WHERE d=? LIMIT 1", 2)}).Into("table1").ToBoundSQL()
        assert.NoError(t, err)
        assert.EqualValues(t, "INSERT INTO table1 (c,d) Values (1,(SELECT b FROM t WHERE d=2 LIMIT 1))", sql)
```
```
$ go test -v -run TestBuilderInsert
=== RUN   TestBuilderInsert
--- FAIL: TestBuilderInsert (0.00s)
    builder_insert_test.go:20: 
        	Error Trace:	builder_insert_test.go:20
        	Error:      	Not equal: 
        	            	expected: "INSERT INTO table1 (c,d,e) Values (1,2,3)"
        	            	actual  : "INSERT INTO table1 (c,d,e) Values (1,3,2)"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-INSERT INTO table1 (c,d,e) Values (1,2,3)
        	            	+INSERT INTO table1 (c,d,e) Values (1,3,2)
        	Test:       	TestBuilderInsert
FAIL
exit status 1
FAIL	github.com/go-xorm/builder	0.006s
```

--- a/builder.go
+++ b/builder.go
@@ -245,6 +245,23 @@ func (b *Builder) Or(cond Cond) *Builder {
 	return b
 }
 
+type insertColsSorter struct {
+	cols []string
+	vals []interface{}
+}
+
+func (s insertColsSorter) Len() int {
+	return len(s.cols)
+}
+func (s insertColsSorter) Swap(i, j int) {
+	s.cols[i], s.cols[j] = s.cols[j], s.cols[i]
+	s.vals[i], s.vals[j] = s.vals[j], s.vals[i]
+}
+
+func (s insertColsSorter) Less(i, j int) bool {
+	return s.cols[i] < s.cols[j]
+}
+
 // Insert sets insert SQL
 func (b *Builder) Insert(eq ...interface{}) *Builder {
 	if len(eq) > 0 {
@@ -275,10 +292,10 @@ func (b *Builder) Insert(eq ...interface{}) *Builder {
 	}
 
 	if len(b.insertCols) == len(b.insertVals) {
-		sort.Slice(b.insertVals, func(i, j int) bool {
-			return b.insertCols[i] < b.insertCols[j]
+		sort.Sort(insertColsSorter{
+			cols: b.insertCols,
+			vals: b.insertVals,
 		})
-		sort.Strings(b.insertCols)
 	}
 	b.optype = insertType
 	return b

--- a/builder_insert_test.go
+++ b/builder_insert_test.go
@@ -15,6 +15,10 @@ func TestBuilderInsert(t *testing.T) {
 	assert.NoError(t, err)
 	assert.EqualValues(t, "INSERT INTO table1 (c,d) Values (1,2)", sql)
 
+	sql, err = Insert(Eq{"e": 3}, Eq{"c": 1}, Eq{"d": 2}).Into("table1").ToBoundSQL()
+	assert.NoError(t, err)
+	assert.EqualValues(t, "INSERT INTO table1 (c,d,e) Values (1,2,3)", sql)
+
 	sql, err = Insert(Eq{"c": 1, "d": Expr("SELECT b FROM t WHERE d=? LIMIT 1", 2)}).Into("table1").ToBoundSQL()
 	assert.NoError(t, err)
 	assert.EqualValues(t, "INSERT INTO table1 (c,d) Values (1,(SELECT b FROM t WHERE d=2 LIMIT 1))", sql)

